cloud/edgecontroller/hack/update-codegen.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../../.. && pwd)
+
+${SCRIPT_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh "deepcopy" \
```

Comment:
right now , we are only doing deep copy , after this , we can add informer , lister , clientset etc , and use those instead of manually creating the crd client for device apis.

Reply:
Sounds fair.